src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -671,72 +603,36 @@ def _prepare_deployment_properties_unmodified(cli_ctx, template_file=None, templ
-
-    template_param_defs = template_obj.get('parameters', {})
-    template_obj['resources'] = template_obj.get('resources', [])
-    parameters = _process_parameters(template_param_defs, parameters) or {}
-    parameters = _get_missing_parameters(parameters, template_obj, _prompt_for_parameters)
-
-    template = json.loads(json.dumps(template))
-    parameters = json.loads(json.dumps(parameters))
-
-    properties = DeploymentProperties(template=template, template_link=template_link,
-                                      parameters=parameters, mode=mode, on_error_deployment=on_error_deployment)
-
-    return properties
-
-
-def _get_deployment_management_client(cli_ctx, handle_extended_json_format=False,
-                                      aux_subscriptions=None, aux_tenants=None):
+def _get_deployment_management_client(cli_ctx, aux_subscriptions=None, aux_tenants=None):
     smc = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES, aux_subscriptions=aux_subscriptions,
                                   aux_tenants=aux_tenants)
+
     deployment_client = smc.deployments  # This solves the multi-api for you
 
-    if handle_extended_json_format:
-        # pylint: disable=protected-access
-        deployment_client._serialize = JSONSerializer(
-            deployment_client._serialize.dependencies
-        )
-
-        # Plug this as default HTTP pipeline
-        from msrest.pipeline import Pipeline
-        from msrest.pipeline.requests import (
-            RequestsCredentialsPolicy,
-            RequestsPatchSession,
-            PipelineRequestsHTTPSender
-        )
-        from msrest.universal_http.requests import RequestsHTTPSender
-
-        smc.config.pipeline = Pipeline(
-            policies=[
-                JsonCTemplatePolicy(),
-                smc.config.user_agent_policy,
-                RequestsPatchSession(),
-                smc.config.http_logger_policy,
-                RequestsCredentialsPolicy(smc.config.credentials)
-            ],
-            sender=PipelineRequestsHTTPSender(RequestsHTTPSender(smc.config))
-        )
+    # pylint: disable=protected-access
+    deployment_client._serialize = JSONSerializer(
+        deployment_client._serialize.dependencies
+    )
+
+    # Plug this as default HTTP pipeline
+    from msrest.pipeline import Pipeline
+    from msrest.pipeline.requests import (
+        RequestsCredentialsPolicy,
+        RequestsPatchSession,
+        PipelineRequestsHTTPSender
+    )
+    from msrest.universal_http.requests import RequestsHTTPSender
+
+    smc.config.pipeline = Pipeline(
+        policies=[
+            JsonCTemplatePolicy(),
+            smc.config.user_agent_policy,
+            RequestsPatchSession(),
+            smc.config.http_logger_policy,
+            RequestsCredentialsPolicy(smc.config.credentials)
+        ],
+        sender=PipelineRequestsHTTPSender(RequestsHTTPSender(smc.config))
+    )
```

Comment:
I will prefer to abstract this part as a new "config client" function.

Reply:
Actually, I just deleted the logic of `if handle_extended_json_format`.  
So I didn't understand your suggestion, Could you please be more specific